lib/ansible/module_utils/basic.py

```diff
@@ -389,7 +392,7 @@ def __init__(self, argument_spec, bypass_checks=False, no_log=False,
         self.required_one_of = required_one_of
         self.required_if = required_if
         self.required_by = required_by
-        self.cleanup_files = []
+        self.cleanup_files: list[str | bytes | os.PathLike] = []
```

Comment:
I think this should be something like
```suggestion
        self.cleanup_files: list[bytes] | list[str] | list[os.PathLike[bytes]] | list[os.PathLike[str]] = []
```
without a union type for the items.

Reply:
Uhm, why? You can add all four different types during the same module run.

Reply:
But probably shouldn't? So why not make MyPy encourage the end-users to use same types?

Reply:
You cannot do that with Python typing. Which type would you add to `AnsibleModule.add_cleanup_file()`'s `path` parameter then?